### DISCUSSION TITLE
How do you handle JSON-LD with Vue3 and Inertiajs ?

### DISCUSSION THREAD

**Discussion by stormtomz0** [inertiajs/inertia#1424] [Help (Vue)] [answered]
Hello,

Is there a way to use JSON-LD with Inertiajs and Vue3 so that it works both clientside and SSR ?

Here is an example from fresh Laravel project with Breeze Vue+SSR starterkit.

```
<Head>
   <component is="script" type="application/ld+json"> {"foo":"bar"} </component>
</Head>
```

After running `npm run build && php artisan inertia:start-ssr` it does show up when you click "view page source" but when you "inspect" the head section there is only empty script tags. So it gets removed in hydration.

- SSR works
- client side does not work
- 'Google's Rich Results Test' does not detect the metadata

However when I run `npm run dev` it shows up in inspect tool.

When I remove the Head tags and append it to the Body section the situation gets kinda flipped.

- SSR does not work (script content is URL-encoded)
- client side works after hydration
- 'Google's Rich Results Test' can read the metadata

However it would be cool to get it working in both and preferably add it into the Head section.

It's possible to teleport that script into the head section but it won't work on SSR.

I would really appreciate any ideas on this. 

JSON-LD is much more convinient than Microdata. 
Spatie has also great 'schema-org' package for it.

**Comment by michaelfromtheoutfit** [comment]
Hey @stormtomz0 - I'm running into the EXACT issue you mention above. Did you ever find a solution?

**Comment by cmercado93** [comment]
I don't know if this helps, mainly because of the time that has passed. What I did to make it work both on SSR and on the client side with Google is to create the Head tag when I'm in SSR. Then, when it renders on the client, I create a Teleport tag that points to the head.

This is an example of a project I'm working on.


```vue
<script setup>
    import { computed } from 'vue';

    const inSSR = computed(() => typeof window == 'undefined');

    ...
</script>
<template>
    <Head v-if="inSSR">
        <component is="script" type="application/ld+json">
            {{
                $seo.makeProduct({
                    name,
                    urlImage,
                    description,
                    urlOffer,
                    price,
                    priceCodeIso,
                })
            }}
        </component>
    </Head>
    <template v-else>
        <Teleport to="head">
            <component is="script" type="application/ld+json">
                {{
                    $seo.makeProduct({
                        name,
                        urlImage,
                        description,
                        urlOffer,
                        price,
                        priceCodeIso,
                    })
                }}
            </component>
        </Teleport>
    </template>
</template>

```

**Comment by stormtomz0** [comment, marked as answer]
This has been fixed in v1.0.9.
